features/370-auth-ui/design.md

```diff
@@ -0,0 +1,393 @@
+There is no validation to enforce this restriction.
+
+#### Customizing with own templates
+
+The developer can also provide their own templates via [the existing template mechanism](../251-templates/index.md).
+
+## The authorize endpoint
+
+The authorize endpoint `/_auth/_/authorize` allows the user to authenticate with configured authentication methods. It handles MFA challenge during authentication. If the user does not have an account yet, the user can sign up directly.
+
+### The authorization page
+
+The authorization page is the landing page of the authorize endpoint. It lists out the configured IdPs. It shows a text field for login ID. The login ID field is either a plain text input or a phone number input, depending on the type of the first login ID key.
+
+```
+|---------------------------|
+| Login with Google         |
+|---------------------------|
+| Login with Facebook       |
+|---------------------------|
+
+              Or
+
+|--------------------------------------|  |----------|
+| Enter in your email or username here |  | Continue |
+|--------------------------------------|  |----------|
+
+Login with a phone number instead.
+```
+
+If the user selects an IdP, the user is redirected to the authorization endpoint of that IdP. If the user continues with a login ID, the user is directed to the password page.
+
+Separating the login ID and the password into two pages allow future improvements, such as [Open ID Provider Issuer Discovery](https://openid.net/specs/openid-connect-discovery-1_0.html#IssuerDiscovery). For example, the user can type in their Google account name and then be redirected to the authorization endpoint of Google.
+
+### The signup page
+
+The signup page displays the first login ID key default. Other login ID keys are available to choose.
+
+```
+Sign with email
+|--------------------------|  |----------|
+| Enter in your email here |  | Continue |
+|--------------------------|  |----------|
+
+Sign with phone instead.
+Sign with username instead.
+```
+
+### The signup password page
+
+The signup password page displays a visibility toggleable password field with password requirements.
+
+```
+|------------------------|  |----------|
+| Create a password here |  | Continue |
+|------------------------|  |----------|
+
+- [ ] At least one digit
+- [ ] At least one uppercase English character
+- [ ] At least one lowercase English character
+- [ ] At least one symbols ~`!@#$%^&*()-_=+[{]}\|;:'",<.>/?
+- [ ] At least 8 characters long
+```
+
+### The verification page
+
+The verification page allows the user to verify their login ID. This page requires authenticated user. Unauthenticated access is redirected to the authorization page. Verification is not enforced by the authentication flow. For example, if auto send verification email is enabled, the email is sent behind the scene and the user is redirected to the application immediately after authentication success.
```

Comment:
I think this may need to be a documented entry point to Auth UI, since developers may want to use it in email templates.